src/dynlib/dynlib_windows.c

```diff
@@ -37,6 +37,46 @@ static char *generate_file_name(const char *name, uint32_t major_ver, uint32_t m
     return versioned_file_name;
 }
 
+static DLL_DIRECTORY_COOKIE add_current_module_to_search()
+{
+    wchar_t path[MAX_PATH];
+    HMODULE hModule = NULL;
+
+    if (GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS, (LPCTSTR)add_current_module_to_search, &hModule) == 0)
+    {
+        LOG_WARNING("Failed to get current module (%d).", GetLastError());
+        return NULL;
+    }
+
+    if (GetModuleFileNameW(hModule, path, sizeof(path)) == 0)
+    {
+        LOG_WARNING("Failed to get current module file name (%d).", GetLastError());
+        return NULL;
+    }
+
+    // GetModuleFileName give the full path, but AddDllDirectory requires a directory path. Remove
+    // the filename portion of this path. This should only be running in the "K4A.dll" assembly,
+    // but verify this assumption before truncating the path. If this code is used from a different
+    // file, then this needs to be modified to search for the path separators.
+    size_t length = wcslen(path);
+    wchar_t *fileName = path + (length - 7);
+    if (fileName <= path || _wcsicmp(fileName, L"k4a.DLL") != 0)
+    {
+        LOG_WARNING("The file name of the current module is not expected.", 0);
+        return NULL;
+    }
+
+    fileName[0] = L'\0';
+
+    DLL_DIRECTORY_COOKIE dllDirectory = AddDllDirectory(path);
```

Comment:
Can you add a comment here explaining why we need to do this.

```suggestion
    // This adds the directory of the current module (k4a.dll) to the loader's path.
    // This will mimic how C# loads DLLs. The loader for C code only loads from the path of the current executable
    DLL_DIRECTORY_COOKIE dllDirectory = AddDllDirectory(path);
```